pkg/messageFormat/ast.go

```
@@ -0,0 +1,101 @@
+package messageFormat
+
+import (
+	"fmt"
+	"strings"
+)
+
+type NodeMatch struct {
+	Selectors []NodeExpr
+	Variants  []NodeVariant
+}
+
+type NodeVariable struct {
+	Name string
+}
+
+type NodeLiteral struct {
+	Value string
+}
+
+type NodeText struct {
+	Text string
+}
+
+type NodeExpr struct {
+	Value    interface{} // NodeVariable | NodeLiteral
+	Function NodeFunction
+}
+
+type NodeVariant struct {
+	Keys    []string
+	Message []interface{}
+}
+
+type NodeFunction struct {
+	Name string
+}
+
+func TokensToMessageFormat(tokens []Token) (NodeMatch, error) {
+	var (
```

what happens if tokens slice is empty? Maybe return an error?